heroku/cli
# refactor(pg-v5): Move command 'pg:kill' to CLI

## Description

Moves the `pg:kill` command to CLI, converting to OClif Core and TS.

## Test instructions

You can test by pulling this branch, installing dependencies with `yarn` and building with `yarn build`.

Then you will need to generate a PostgreSQL backend process or query that you can 'kill'. I've tested by opening an interactive `psql` session to the database using `heroku pg:psql` and starting a transaction with `BEGIN TRANSACTION;` (do this in a second terminal window). Then you can fetch the PID linked with that backend process with `heroku pg:ps` and use it to kill it. When you `--force` kill it, you should see that the connection to `psql` is lost and then reset on the secondary terminal window.

```
./bin/run pg:kill --help                           # always test that help looks ok.
./bin/run pg:kill PID -a mytest-app                # attempts to cancel the query running on PID for primary database.
./bin/run pg:kill PID -a mytest-app --force        # attempts to kill the process PID for primary database.
```

You can pass a second DATABASE argument to run the command against a specific database.

Compare outputs against the current CLI.

## SOC2 Compliance
[GUS Work Item](https://gus.lightning.force.com/a07EE00001bDJofYAG) (Salesforce Internal)
